### ISSUE TITLE
InputText problem

### ISSUE THREAD

**Issue by mavmarong** [mellinoe/ImGui.NET#317]
I can't type in my InputText:

**Comment by Zarickan** [comment]
This is how I do it, which works without any issues so far. Suppose you want to edit a string stored in `value` (which might be null):
```c#
var str = value ?? "";       
ImGui.PushID(id);
ImGui.InputText(label, ref str, 1024, ImGuiInputTextFlags.CallbackAlways, data =>
{
    str = Marshal.PtrToStringUTF8((IntPtr) data->Buf) ?? string.Empty;
    return 1;
});
ImGui.PopID();
```

For longer strings this needs to be adjusted, if the label of the input is unique then there is no reason to Push/Pop an id either.

**Comment by mavmarong** [comment]
Marshal doesn't contain the method "PtrToStringUTF8"

**Comment by mavmarong** [comment]
Marshal

> This is how I do it, which works without any issues so far. Suppose you want to edit a string stored in `value` (which might be null):
> 
> ```cs
> var str = value ?? "";       
> ImGui.PushID(id);
> ImGui.InputText(label, ref str, 1024, ImGuiInputTextFlags.CallbackAlways, data =>
> {
>     str = Marshal.PtrToStringUTF8((IntPtr) data->Buf) ?? string.Empty;
>     return 1;
> });
> ImGui.PopID();
> ```
> For longer strings this needs to be adjusted, if the label of the input is unique then there is no reason to Push/Pop an id either.

Marshal doesnt contain the method named "PtrToStringUTF8".

**Comment by Zarickan** [comment]
Are you on .NET Framework? I think `Marshal.PtrToStringUTF8` was added at some point in .net standard/.net core. If you are working with only ansi characters `Marshall.PtrToStringAnsi` should do the trick.

**Comment by mavmarong** [comment]
> Are you on .NET Framework? I think `Marshal.PtrToStringUTF8` was added at some point in .net standard/.net core. If you are working with only ansi characters `Marshall.PtrToStringAnsi` should do the trick.
the compiler gives me this error because of "data" https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs0214?f1url=%3FappId%3Droslyn%26k%3Dk(CS0214)

**Comment by Zarickan** [comment]
Yes, the code is unsafe, you have to wrap it in an unsafe block (or an unsafe function, I have it in an unsafe helper function):
```cs
public static unsafe string String(string id, string label, string value)
{
    var str = value ?? "";
    
    ImGui.PushID(id);
    ImGui.InputText(label, ref str, 1024, ImGuiInputTextFlags.CallbackAlways, data =>
    {
        str = Marshal.PtrToStringUTF8((IntPtr) data->Buf) ?? string.Empty;
        return 1;
    });
    ImGui.PopID();

    return str;
}
```

**Comment by CEO-Chair** [comment]
see #327, it's caused by an incorrect calling convention
